### PR TITLE
Add PostgreSQL chart.

### PR DESCRIPTION
**Pull request by mtricht**

### CODE REVIEW

**Comment by prydonius** [comment]
@k8s-bot ok to test

File: stable/postgresql/templates/NOTES.txt [hunk 1]
```diff
@@ -0,0 +1,11 @@
+PostgreSQL can be accessed via port 5432 on the following DNS name from within your cluster:
+{{ template "fullname" . }}.{{ .Release.Namespace }}.svc.cluster.local
+
+To get your user password run:
+
+    printf $(printf '\%o' `kubectl get secret --namespace {{ .Release.Namespace }} {{ template "fullname" . }} -o jsonpath="{.data.postgres-password[*]}"`);echo
+
+To connect to your database run the following command (using the correct password):
+
+   kubectl run {{ template "fullname" . }}-client --rm --tty -i --image postgres --command -- psql -h {{ template "fullname" . }} {{- if .Values.postgresUser }} -U{{ .Values.postgresUser }}{{ else }} -Upostgres{{- end -}}
```
Comment (prydonius) [inline stable/postgresql/templates/NOTES.txt]: I can't seem to enter the password using this command. Perhaps you can provide it using an ENV var like the patroni chart does: https://github.com/kubernetes/charts/blob/master/incubator/patroni/templates/NOTES.txt#L17

**Comment by prydonius** [comment]
@swordbeta thanks! I'm getting a CrashLoopBackOff error with the following logs:

```
initdb: directory "/var/lib/postgresql/data/pgdata" exists but is not empty
It contains a lost+found directory, perhaps due to it being a mount point.
Using a mount point directly as the data directory is not recommended.
Create a subdirectory under the mount point.
```

File: stable/postgresql/templates/deployment.yaml [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+          value: /var/lib/postgresql/data/pgdata
+        - name: POSTGRES_PASSWORD
+          valueFrom:
+            secretKeyRef:
+              name: {{ template "fullname" . }}
+              key: postgres-password
+        ports:
+        - name: postgresql
+          containerPort: 5432
+        livenessProbe:
```
Comment (chrislovecnm) [inline stable/postgresql/templates/deployment.yaml:59]: Readyprove?

File: stable/postgresql/README.md [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+```
+
+## Introduction
+
+This chart bootstraps a [PostgreSQL](https://github.com/docker-library/postgres) deployment on a [Kubernetes](http://kubernetes.io) cluster using the [Helm](https://helm.sh) package manager.
+
+## Prerequisites
+
+- Kubernetes 1.4+ with Beta APIs enabled
+- PV provisioner support in the underlying infrastructure
```
Comment (chrislovecnm) [inline stable/postgresql/README.md]: If you do an empty doe do you need a PV provisioner? Maybe change the language for that. Local dev I could run on minikube with emptydir

**Review by chrislovecnm — CHANGES_REQUESTED** [review]
Looks great, just a couple of comments. You an emacs user?

File: stable/postgresql/templates/deployment.yaml (hunk 2 above)
Comment (chrislovecnm) [reply stable/postgresql/templates/deployment.yaml:59]: readyprobe ... Frigin autocomplete on mobile

**Comment by viglesiasce** [comment]
@swordbeta This is what I did in the mysql chart for the lost+found issue:
https://github.com/kubernetes/charts/blob/master/stable/mysql/templates/deployment.yaml#L16

File: stable/postgresql/templates/NOTES.txt (hunk 1 above)
Comment (mtricht) [reply stable/postgresql/templates/NOTES.txt]: @prydonius You can it's just really unclear: (I pressed enter and entered no password)

```
Waiting for pod default/ok-postgresql-client-2287579111-zo7qi to be running, status is Pending, pod ready: false
If you don't see a command prompt, try pressing enter.

psql: fe_sendauth: no password supplied
```

Changed it to patroni's style.

File: stable/postgresql/templates/deployment.yaml (hunk 2 above)
Comment (mtricht) [reply stable/postgresql/templates/deployment.yaml:59]: @chrislovecnm The `readinessProbe` is a few lines below this one!

And I don't use emacs, what makes you think that haha?

File: stable/postgresql/README.md (hunk 3 above)
Comment (mtricht) [reply stable/postgresql/README.md]: @chrislovecnm I copied this from one of the stable READMEs (actually all of them say this). Changed it nevertheless.

**Comment by mtricht** [comment]
@prydonius I haven't been able to reproduce the `lost+found` issue, but I have added an init container like @viglesiasce suggested which deletes it. Can you let me know if it works?

@viglesiasce The `mysql` annotation uses `pod.alpha.kubernetes.io/init-containers`, shouldn't alpha features be omitted from the stable graphs? (I used `pod.beta.kubernetes.io/init-containers`; I'm not sure if there's any difference. The [documentation](http://kubernetes.io/docs/user-guide/production-pods/#handling-initialization) mentions it's in beta but the example uses alpha.)

File: stable/postgresql/templates/deployment.yaml (hunk 2 above)
Comment (chrislovecnm) [reply stable/postgresql/templates/deployment.yaml:59]: Missed a EOF on one of the file

Comment (chrislovecnm) [reply stable/postgresql/templates/deployment.yaml:59]: Thanks ... Missed thay

Comment (chrislovecnm) [reply stable/postgresql/templates/deployment.yaml:59]: So they have changed edit on these. My engrish is sucking tonight. So there was an EOF marker that was missed on on of the files. One of my co-workers gets that with emacs all the time.

Missed your probes, you rock!

**Review by chrislovecnm — APPROVED** [review]
LGTM

**Comment by chrislovecnm** [comment]
@swordbeta beta vs alpha depends on the version of k8s you are running. I am wondering if they should be vars to allow for easy upgrade. Another topic that we need to talk about.

**Comment by viglesiasce** [comment]
@swordbeta @chrislovecnm the feature is now in beta and the annotation as listed continues to work. We should look to update these across the board soon but it is not currently necessary.

**Comment by chrislovecnm** [comment]
I have an issue open Vic

**Pull request merged**